lib/ReactViews/Custom/Chart/MomentPointsChart.jsx

```diff
@@ -31,36 +33,37 @@ class MomentPointsChart extends React.Component {
         domain: basisItemScales.y.domain(),
         range: scales.y.domain()
       });
-      const interpolatedPoints = chartItem.points.map(p =>
-        interpolate(p, basisItem.points, basisToSourceScale)
-      );
+      const interpolatedPoints = chartItem.points.map(p => ({
+        ...p,
+        ...interpolate(p, basisItem.points, basisToSourceScale)
+      }));
       return interpolatedPoints;
     }
     return chartItem.points;
   }
 
   doZoom(scales) {
-    const points = this.props.chartItem.points;
-    if (points.length === 0) return;
-    const n = points.length;
-    const spacing = (scales.x(points[n - 1].x) - scales.x(points[0].x)) / n;
-    const dots = document.querySelectorAll(`g#${this.props.id} circle`);
-    dots.forEach((dot, i) => {
+    const points = this.points;
+    if (points.length === 0) {
+      return;
+    }
+    const glyphs = document.querySelectorAll(
```

Comment:
this bit seems weird to me in that we are reaching into the DOM to modify things instead of using React but i recognise that this pattern is used by all the different charting components and will be a big task to migrate over

Reply:
The reason for modifying DOM directly is performance. Because we chart many thousands of points, any updates to those points (like when you zoom the chart) results in react destroying and re-creating DOM elements for those points. This is painfully slow. Instead what we do here is after react has initially rendered those (svg) DOM elements and when we have to update the position of those points, we just change the attributes of the existing DOM elements. There is a drastic improvement in performance when we do that.